src/frontend/get-github-setup.ts

```diff
@@ -14,17 +14,39 @@ export default async (req: Request, res: Response): Promise<void> => {
 	req.log.info("Received get github setup page request");
 	const { jiraHost } = res.locals;
 
+	const installationId = req.originalUrl.split("=")[1].split("&")[0]
 	let redirectUrl = getJiraMarketplaceUrl(jiraHost);
 
+	const {	github, client } = res.locals;
+
+	const { data: { installations } } = await github.apps.listInstallationsForAuthenticatedUser();
+	const { data: info } = await client.apps.getAuthenticated();
+
+	const installation = installations.filter((item) => item.id === Number(installationId));
+
 	// If we know enough about user and site, redirect to the app
 	if (jiraHost && await jiraSiteExists(jiraHost) && await Installation.getForHost(jiraHost)) {
 		redirectUrl = getJiraAppUrl(jiraHost);
 	}
 
+	let jiraInstallation;
+
+	if (jiraHost) {
+		jiraInstallation = await Installation.getForHost(jiraHost);
+	}
+
+	const hasJiraHost = !!jiraHost;
+
 	res.render("github-setup.hbs", {
 		csrfToken: req.csrfToken(),
 		nonce: res.locals.nonce,
 		jiraHost,
-		redirectUrl
+		redirectUrl,
+		hasJiraHost,
+		clientKey: jiraInstallation?.clientKey,
+		orgName: installation[0]?.account?.login,
+		avatar: installation[0]?.account?.avatar_url,
```

Comment:
Naming an array in singular is a bit odd. I'd also rather assign the installation to a constant instead of reading it from the error twice here. Then you can also throw an error if the array is empty for whatever reason and remove the `?` here, because is it really a valid state when the `installation` is `undefined`?